src/rospack_backcompat.cpp

```diff
@@ -40,7 +40,8 @@ namespace rospack
 int
 ROSPack::run(int argc, char** argv)
 {
-  rospack::Rospack rp;
+  // allow caching of results between calls (of same process)
+  static rospack::Rospack rp;
```

Comment:
As far as I could see, the rospack re-crawls every time when the `search_path` [fetched from the environment](https://github.com/rhaschke/rospack/blob/e1bcabef5955bb7377474ff5c45a5b9bedaaa104/src/rospack_cmdline.cpp#L195) has [changed](https://github.com/rhaschke/rospack/blob/c6dc1b4914886f135c69c5ef9eb6d08243097a5c/src/rospack.cpp#L344-L363). Hence, there is no need to pass in a new rospack instance, isn't it?

If I'm wrong, where would you pass in the rospack instance? In [roslib](https://github.com/rhaschke/ros/blob/8a12cd02bfa7ec8b0560339a3c1e0c455b763a66/core/roslib/src/package.cpp#L47) and all corresponding functions? That would break the API, which I tried to avoid.